test/integration/thermal_camera.cc

```diff
@@ -256,6 +260,191 @@ void ThermalCameraTest::ThermalCameraBoxes(
   ignition::rendering::unloadEngine(engine->Name());
 }
 
+//////////////////////////////////////////////////
+void ThermalCameraTest::ThermalCameraBoxes8Bit(
+    const std::string &_renderEngine)
+{
+  int imgWidth = 50;
+  int imgHeight = 50;
+  double aspectRatio = imgWidth/imgHeight;
+
+  double unitBoxSize = 1.0;
+  ignition::math::Vector3d boxPosition(1.8, 0.0, 0.0);
+
+  // Only ogre2 supports 8 bit image format
+  if (_renderEngine.compare("ogre2") != 0)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' doesn't support 8 bit thermal cameras" << std::endl;
+    return;
+  }
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+
+  // red background
+  scene->SetBackgroundColor(1.0, 0.0, 0.0);
+
+  // Create an scene with a box in it
+  scene->SetAmbientLight(1.0, 1.0, 1.0);
+  ignition::rendering::VisualPtr root = scene->RootVisual();
+
+  // create box visual
+  ignition::rendering::VisualPtr box = scene->CreateVisual();
+  box->AddGeometry(scene->CreateBox());
+  box->SetOrigin(0.0, 0.0, 0.0);
+  box->SetLocalPosition(boxPosition);
+  box->SetLocalRotation(0, 0, 0);
+  box->SetLocalScale(unitBoxSize, unitBoxSize, unitBoxSize);
+
+  // set box temperature
+  float boxTemp = 310.0;
+  box->SetUserData("temperature", boxTemp);
+
+  root->AddChild(box);
+  {
+    // range is hardcoded in shaders
+    float boxTempRange = 3.0;
+    double farDist = 10.0;
+    double nearDist = 0.15;
+    double hfov = 1.05;
+    double minTemp = 253.0;
+    double maxTemp = 673.0;
```

Comment:
If a 16-bit camera with .01 mK linear resolution has a maximum temperature of `655.35` kelvin, can an 8-bit camera have a maximum temperature that is higher than `655.35`? I'd think that the 8-bit camera would have more of a limited range than a 16-bit camera, but maybe I am misunderstanding the math.

Also, speaking of the math behind this, can a comment be added that explains how the values for `minTemp` and `maxTemp` were calculated?

Reply:
The `minTemp` and `maxTemp` are actually from physical thermal camera specs.

As an example,  the max temperature that can be represented by an 8 bit camera with 3 degree resolution is: 
* (2^bitDepth-1) * resolution = 255*3 = 765 degrees.

Here we have a user-defined max of 673.0. So any values above 673.0 will be clamped even when the image is able to store a higher number.

On the other hand, if user species a max temp of 800, which is above the limit (765) on what the 8 bit 3 resolution camera can produce, we can only return a max of 765 degrees

This FLIR page gave another example: https://www.flir.com/support-center/Instruments/what-is-the-relation-between-temperaturelinearresolution-and-sensorgainmode-temperature-range-for-the-flir-ax5-series/

> Even though the SensorGainMode is set to LowGainMode (–40 to +550°C), the camera signal will saturate around 382°C.

Reply:
Thanks for the explanation. Should we add a note here that these min/max values are from actual physical thermal camera tests, and include a link to the specs in case we have to re-visit this test later?

Reply:
8a5bf29ea77db8d83618bc6bb93a5c8f7f29c16c